ndmg/stats/qa_tensor.py

```diff
@@ -33,86 +34,167 @@
 import sys
 import matplotlib
 
+from dipy.viz import window, actor
+from fury.actor import orient2rgb
+
 matplotlib.use("Agg")  # very important above pyplot import
 import matplotlib.pyplot as plt
 
-
-def tensor2fa(tensors, tensor_name, dti, derivdir, qcdir):
+def generate_3_d_directions(peak_dirs,peak_values):
     """
-    outdir: location of output directory.
-    fname: name of output fa map file. default is none (name created based on
-    input file)
+    Generates 3-d data required for plotting directions for the entire brain volume
+    
+    Parameters
+    -----------
+    peak_dirs: np array of peak_dirs 
+    peak_values: np array of peak_values
+    slices: dictionary of lists of slices in x,y,z plane
+    
+    Returns
+    -----------
+    centers: np array of center points
+    directions: np array of vector directions
+    directions_colors: np array of RGB colors 
+    heights: np array of vector magnitudes
     """
-    dti_data = nb.load(dti)
-    affine = dti_data.get_affine()
-    dti_data = dti_data.get_data()
-
-    # create FA map
-    FA = fractional_anisotropy(tensors.evals)
-    FA[np.isnan(FA)] = 0
-
-    # generate the RGB FA map
-    FA = np.clip(FA, 0, 1)
-    RGB = color_fa(FA, tensors.evecs)
-
-    fname = os.path.split(tensor_name)[1].split(".")[0] + "_fa_rgb.nii.gz"
-    fa = nb.Nifti1Image(np.array(255 * RGB, "uint8"), affine)
-    nb.save(fa, os.path.join(derivdir, fname))
-
-    fa_pngs(fa, fname, qcdir)
-
-
-def fa_pngs(data, fname, outdir):
+    
+    #initialize 
+    centers = []
+    directions = []
+    heights = []
+    directions_colors = []
+    
+    #iteratively create required new arrow entries per value in 3-d peak object
+    for x in range(peak_dirs.shape[0]):
+        for y in range(peak_dirs.shape[1]): 
+            for z in range(peak_dirs.shape[2]):
+                centers.append([x,y,z])
+                
+                #relies on the fact that peaks_from_models generates peak_dirs and peak_values in descending order
+                directions.append(peak_dirs[x,y,z,0,:])
+                heights.append(peak_values[x,y,z,0])
+
+    #convert to np arrays
+    centers = np.asarray(centers)
+    directions = np.asarray(directions)
```

Comment:
`directions` is a list of 1d numpy arrays, which is kind of a weird object. A single 2d numpy array would likely be better. Likely not important enough for you to spend time changing though

Reply:
I'm fine with you not changing this for now, though. Feel free to just resolve this after reading, unless you want to make the change.